### DISCUSSION TITLE
Looking into allowing more customization of the table

### DISCUSSION THREAD

**Discussion by Mohamed-Ayaan358** [langfuse/langfuse#1941] [Ideas]
### Describe the feature or potential improvement

I feel like it would be highly beneficial if one could place new columns inside any of the tables, so that we can input any of the values we would wish the dashboard to contain in addition to the present values.

We could additionally also introduce a method by which one can use to input a value into this column similar to the trace.score() method.

For example If I wished to also add the ground truth as a column in a trace that possessed the query input and response output it would be beneficial.

### Additional information

I wish to try and write this myself could anyone assist me in the matter, I have been able to manually enter a new column into the generations table, however it does not appear on the sorting dropdown (Is this by any chance dependent on the datatype proivided or is an api call to obtain this.)

Additionally does anyone know what table is being used here

**Comment by marcklingen** [comment]
I think breaking out a specific metadata key into it's own column could be the best solution to this. No plans yet though to add this as all table features (sorting, search) will be rather slow on the metadata column for now. [v3](https://github.com/orgs/langfuse/discussions/1902) might change this.

**Reply by Mohamed-Ayaan358** [reply]
Of course, thank you for offering a solution, is it possible for you could state which table library is being used in the repository site?
I am currently parsing through the code.

**Reply by marcklingen** [reply]
shadcn/ui & tanstack react-table
